radio/src/targets/common/arm/stm32/rtc_driver.cpp

```diff
@@ -22,6 +22,57 @@
 #include "stm32_hal.h"
 #include "rtc.h"
 
+//
+// Color screen targets use the first 2 out of the 20 
+// RTC backup registers to hold data about startup reason 
+// and type.
+
+//
+// Calculates the RTC backup memory checksum and stores it
+// in the last backup register location (BKP19R).
+//
+#define RTCCHKSUM() ((RTC->BKP0R ^ RTC->BKP1R) + 45444745)
+
+//  
+// resets RTC backup registers if data is corrupt
+// and returns the requested backup register
+// 
+uint32_t getRTCBKPR(uint8_t RTCBKPRegister) {
+  uint32_t prim = __get_PRIMASK();
+  
+  __disable_irq();
+  
+  if(RTC->BKP19R != RTCCHKSUM()) {
+    RTC->BKP0R = 0;
+    RTC->BKP1R = 0;
+    RTC->BKP19R = RTCCHKSUM();
+  }
+
+  uint32_t value = ((uint32_t *)RTCBKBR_BASE)[RTCBKPRegister];   // RTC backup registers are 32bit registers
+
+  if(!prim) 
+    __enable_irq();
```

Comment:
Are you sure it is absolutely required to disable interrupts while settings this? Is this really accessed from different RTOS tasks?

Reply:
boardOff() is called from various places, e.g. GUI task. The main intention was to make reading/writing backup registers and calculating the checksum atomic for data consistency. Does it do any harm?

Reply:
Normally you would use the FreeRTOS critical section, as disabling all IRQs will effectively prevent FreeRTOS from doing anything, with unpredictable consequences. Disabling IRQs should only be used when there is no other solution.

Reply:
please advise on which RTOS call is appropriate and how to choose the correct include files.